galsim/compound.py

```diff
@@ -322,18 +322,26 @@ def __init__(self, *args, **kwargs):
             else:
                 real_space = False
 
-        # Warn if doing DFT convolution for objects with hard edges.
+        # Warn if doing DFT convolution for objects with hard edges
         if not real_space and hard_edge:
-            import warnings
-            if len(args) == 2:
-                msg = """
-                Doing convolution of 2 objects, both with hard edges.
-                This might be more accurate and/or faster using real_space=True"""
-            else:
-                msg = """
-                Doing convolution where all objects have hard edges.
-                There might be some inaccuracies due to ringing in k-space."""
-            warnings.warn(msg)
+            
+            # If any aren't analytic, we don't have a choice here, so suppress the warning
```

Comment:
I could make this change, but it still leaves open the possibility that an issue could arise from a convolution of three or more profiles. Eg., convolving:

A - Not real-space analytic
B - Hard edges, real-space analytic
C - Hard edges, real-space analytic

This isn't likely to arise (it would require eg. a PSF with hard edges), but it's possible. In fact, I think I did something almost like this myself once for other work, except A was real-space analytic so problems didn't arise. But if it hadn't have been, a warning would have been raised no matter what I tried to do. So, I think it is proper to put this check here.